pkg/controllers/networkinfo/networkinfo_utils.go

```diff
@@ -130,6 +130,26 @@ func setVPCNetworkConfigurationStatusWithSnatEnabled(ctx context.Context, client
 	}
 }
 
+func setVPCNetworkConfigurationStatusWithNoExternalIPBlock(ctx context.Context, client client.Client, nc *v1alpha1.VPCNetworkConfiguration, hasExternalIPs bool) {
+	newCondition := v1alpha1.Condition{
+		Type:               v1alpha1.ExternalIPBlocksConfigured,
+		LastTransitionTime: metav1.Time{},
+	}
+	if !hasExternalIPs {
+		newCondition.Status = v1.ConditionFalse
+		newCondition.Reason = "No External IP Blocks exist in VPC Connectivity Profile"
```

Comment:
This looks more like a `newCondition.Message` than a `newCondition.Reason`... @dantingl is it by design?

Reply:
@wenqiq could you follow previous condition reason to add it here
https://github.com/vmware-tanzu/nsx-operator/blob/0c41f249845f95b089012407008c7efd2da3b21e/pkg/nsx/services/common/types.go#L177

Reply:
Should use the previous design or change the reason msg?
```
- type: ExternalIPBlocksConfigured
    status: "False"
    reason: "No External IP Blocks exist in VPC Connectivity Profile"
```